shell.nix

```diff
@@ -21,11 +23,22 @@ let
       }
     '';
   });
-  withAuxx = pkgs.runCommand "shell" {
+  withAuxx = pkgs.mkShell {
+    name = "io-with-auxx";
     buildInputs = [
       iohk-ops
       iohkpkgs.cardano-sl-auxx
       iohkpkgs.cardano-sl-tools
     ];
-  } "echo use nix-shell";
+    shellHook = ''
+      ${ioShellHook}
+      for prog in cardano-keygen cardano-x509-certificates cardano-auxx; do
```

Comment:
can't we pull them out of the buildInputs somehow?

Reply:
We could have, except that I added a program to cardano-sl-tools which doesn't use optparse-applicative...